core/Archive/ArchivePurger.php

```diff
@@ -147,6 +147,55 @@ public function purgeOutdatedArchives(Date $dateStart)
         return $deletedRowCount;
     }
 
+    /**
+     * Removes the broken archives for the given month
+     * (meaning they are not marked with a Done flag correctly)
+     *
+     * @param Date $dateStart The date to purge from
+     * @param Date|null $dateEnd The date to purge up to, if not provided then will purge to end of the same month as $dateStart.
+     *                  if $dateEnd is in the past relative to $dateStart, it is ignored
+     * @return int Returns the total number of rows deleted.
+     */
+    public function purgeBrokenArchives(Date $dateStart, Date $dateEnd = null): int
+    {
+        if (!isset($dateEnd) || $dateStart->isLater($dateEnd)) {
+            $dateEnd = $dateStart->getEndOfMonth();
+        }
+
+        $monthStart = $dateStart->getStartOfMonth();
+        $monthEnd = $dateStart->getEndOfMonth();
+
+        $numRowsDeleted = 0;
+
+        // loop through months and delete from relevant tables
+        do {
+            $start = $dateStart->isLater($monthStart) ? $dateStart : $monthStart;
+            $end = $dateEnd->isLater($monthEnd) ? $monthEnd : $dateEnd;
+            $idArchivesToDelete = $this->getBrokenArchiveIds($start, $end);
+            $deletedRowCount = 0;
+            if (!empty($idArchivesToDelete)) {
+                $deletedRowCount = $this->deleteArchiveIds($dateStart, $idArchivesToDelete);
+
+                $this->logger->info("Deleted {count} rows in archive tables (numeric + blob) for {date}.", array(
```

Comment:
Are we only deleting from the numeric tables or blobs as well?
```suggestion
                $this->logger->info("Deleted {count} rows in archive numeric tables for {date}.", array(
```

Reply:
`deleteArchiveIds()` deletes from both archive and blob tables, and these broken archives appear in both tables so we need to delete from both.

Reply:
Fair enough. Was just wondering as some of the other messages only mention numeric tables, perhaps need to adjust those instead?

Reply:
well we are only searching the numeric table for broken archives, thats where that comes from in the other logs. we are just expecting any broken archives to also be broken in the blob tables